### PR TITLE
feat!: Upgrade MySQL2 to latest version

### PR DESCRIPTION
**Pull request by ZigaStrgar**
## Changes

* Upgrades `mysql2` version to `^3.9.7`

**TL;DR**

I don't believe anything is breaking change in terms of `db-migrate-mysql`, locally all 44 tests are passing using this branch. None the less, to stay on the safe side, I'd release this as `v3` of `db-migrate-mysql`.

### Notable changes from v2 to v3 - mysql2

* CI
  * Usage of release-please
  * Node v18 as default
* Community files
  * Code of conduct
  * Readme
  * Typos fix
  * Security.md
  * Typings
* Updated `.d.ts` files
* New exports in `index.js`
  * [One](https://github.com/sidorares/node-mysql2/compare/v2.3.3...v3.0.0#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R15)
  * [Two](https://github.com/sidorares/node-mysql2/compare/v2.3.3...v3.0.0#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R46)
* Added support for two new auth plugins
  * `mysql_clear_password`
  * `mysql_native_password`
  * [Source](https://github.com/sidorares/node-mysql2/compare/v2.3.3...v3.0.0#diff-29b310f2f6d67bc1e32c97bfedf2bdd492811adea75e74040f136f8f5b006a6bR17-R18)
* [Added MFA support](https://github.com/sidorares/node-mysql2/compare/v2.3.3...v3.0.0#diff-52f1bcec70eee1a79ccfc9857584075d7c51b87c402a6dcd9de11ea8b9e5cde9R50-R56) in client handshake
* Few [changes](https://github.com/sidorares/node-mysql2/compare/v2.3.3...v3.0.0#diff-7221fea000805831aadf022ad9b6ebceeb00df99033a5a9787d43279cde5d9c7R10-R22) in `Quit` command - On first glance did not find usages of it
* Added [more charset](https://github.com/sidorares/node-mysql2/compare/v2.3.3...v3.0.0#diff-bea9bc9140aae589b318cf4a0391cea9835436fc93db1b77ddd70fa497f81e7cR226-R275) constants
* Added [reverse map](https://github.com/sidorares/node-mysql2/compare/v2.3.3...v3.0.0#diff-2f203a32c2d5029265eb64cbc46be328caf2a9ed4911bda2f58ebe8d9c36c9f3R3-R32) for types
* [Implementation](https://github.com/sidorares/node-mysql2/compare/v2.3.3...v3.0.0#diff-74f86159a30bde17927594d05ba9bbcdf58281b7f63fae582429cc74de9afa3fR13-R62) for `toPacket`; No more throw NotImplemented
* [Replace](https://github.com/sidorares/node-mysql2/compare/v2.3.3...v3.0.0#diff-17a31987f7995e11080c24188e0c50f5fbf8436863484273cdb723444da78e5dR177) `throw error` with `return cb(error)`

Taken from this comparison: https://github.com/sidorares/node-mysql2/compare/v2.3.3...v3.0.0#diff-17a31987f7995e11080c24188e0c50f5fbf8436863484273cdb723444da78e5dR177

### CODE REVIEW

**Comment by ZigaStrgar** [comment]
Hey @wzrdtales, would you mind taking this a look? 🙏

**Comment by obriet04** [comment]
Hi. We're running into issues with this critical vulnerability, so the changes in this PR would be a big help to us as well.

**Comment by klansser** [comment]
Do you think it's ready to merge @wzrdtales ? It fixes some critical vulnerabilities that should be fixed a long time ago. I'll be glad if you can merge and tag 🙃

**Comment by wzrdtales** [comment]
I think we can release this, I will release the corresponding driver as a major version though as I can't guarantee nothing breaks.
